src/app/core/services/notifications/fcm-notification.service.ts

```diff
@@ -112,6 +113,13 @@ export abstract class FcmNotificationService extends NotificationService {
       : getSeconds({ minutes: this.ttlMinutes })
   }
 
+  unregisterFromNotificataions(): Promise<any> {
+    // NOTE: This will delete the current device token and stop receiving notifications
+    return this.firebase
+      .setAutoInitEnabled(false)
+      .then(() => this.firebase.unregister())
+  }
```

Comment:
should we update the appserver user and delete the FCM token? Or at least it will need to be updated when the token is refreshed.

Reply:
Yes everytime the app is opened (if FCM_REST is enabled) we check if the user is present in the app server. If the user does not exist, we add and if it doesn't exist, we update with the latest subject data in the app (including the current FCM token).
https://github.com/RADAR-base/RADAR-Questionnaire/blob/88cfb7ab83d3bec7ba7380fb17bbf837e768a730/src/app/core/services/app-server/app-server.service.ts#L116-L122

Reply:
ok great, thanks. Is there a particular reason for this? I think the official docs discourage from changing the token usually.

Reply:
can't we just cancel/delete all the notifications on the server on reset?

Reply:
I was reading through the Firebase plugin docs and saw that they usually unregister whenever a user is logged out of an app. Here: https://github.com/dpa99c/cordova-plugin-firebasex#unregister
Also, just so the token is unique to each subject especially when sharing phones, they tend to overlap?

Reply:
ah ok, yes that makes sense if they are sharing phones. Thanks for clarifying.